src/components/LogoutButton.js

```diff
@@ -16,9 +18,27 @@ const useStyles = makeStyles((theme) => ({
 const LogoutButton = () => {
   const history = useHistory();
   const dispatch = useDispatch();
-  const onClick = async () => {
+  const modulesManager = useModulesManager();
+  const mPassLogout = modulesManager.getConf("fe-core", "LogoutButton.showMPassProvider", false);
+  const onClick = async (e) => {
+    if (mPassLogout) {
+      await redirectToMPassLogout(e);
+    } else {
+      await redirectToImisLogout();
+    }
```

Comment:
Personally, I'm not a big fan of the 'else' block. In this case, it could be skipped.
`
    if (mPassLogout) {
      await redirectToMPassLogout(e);
    };
    await redirectToImisLogout();
`
You can consider a change, it's up to you.